### PR TITLE
[fix] Only injected public repositories

### PR DESCRIPTION
**Pull request by wxharry**
## Purpose

This PR is initiated by https://github.com/hypertrons/hypertrons-crx/issues/276

#### Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Approach

_How does this change address the problem?_

I created a new function `isPublic` to check the visibility of the current repository. The function simply compares the content of the top-left secondary label and return a Boolean. And I added `isPublic` to the `runsWhen` rules of `PerceptorTab` so that the feature would not be loaded on a private repository at `loadFeatures` phase.

### CODE REVIEW

File: src/pages/ContentScripts/Perceptor.tsx [hunk 1]
```diff
@@ -60,7 +60,7 @@ export class Perceptor extends PerceptorBase {
         logger.info(featureId, 'is disabled');
         return;
       }
-      if (Feature.prototype.include.every((c: () => any) => !c())) {
+      if (!Feature.prototype.include.every((c: () => any) => c())) {
```
Comment (tyn1998) [inline src/pages/ContentScripts/Perceptor.tsx:63]: I think this way is more natural, too :)

**Comment by tyn1998** [comment]
LGTM, much thanks! Just one thing left to be checked: if it's true that there is no such API provided by pageDetect to check the repo's visibility?

File: src/pages/ContentScripts/Perceptor.tsx (hunk 1 above)
Comment (wxharry) [reply src/pages/ContentScripts/Perceptor.tsx:63]: The logic is different here. The previous version implemented an `or` logic and in this change it is an `and` logic.

**Comment by wxharry** [comment]
It seems that `pageDetect` only parses URL to give information. Since the URL of the repository page does not carry information on visibility, I don't think it can check the visibility.

File: src/pages/ContentScripts/Perceptor.tsx (hunk 1 above)
Comment (tyn1998) [reply src/pages/ContentScripts/Perceptor.tsx:63]: Ah oh, that is not right. Take an example:
image
The `Hypertrons` is the button injected in both PR & Issue pages:
image

So it should be `or` rather `and`. Maybe another strategy should be made to handle this.

Comment (wxharry) [reply src/pages/ContentScripts/Perceptor.tsx:63]: OK, I will try to figure!

**Review by tyn1998 — APPROVED** [review]

**Review by tyn1998 — COMMENTED** [review]
Great job!

**Comment by tyn1998** [comment]
I just `gh pr checkout 320` and had a run, everything looks fine and works well. Next, please run `yarn run prettier` to format code then do `git rebase` on upstream/master as [CONTRIBUTING](https://github.com/hypertrons/hypertrons-crx/blob/master/CONTRIBUTING.md#7-sync-your-local-repository-with-the-upstream) requires and then I will merge your pull request!

Thanks again!

**Comment by tyn1998** [comment]
Hey wxharry, I just bumpped the version number and made a commit into your PR(branch) by following:
1. `gh pr checkout 320`
2. make changes and a commit
3. `git push`

Then I successfully add a commit to your PR!

Here conflict rises. It's not a big deal. Though I can handle with that by continue adding commits to your PR, but you might want to figure it out by yourself. If you have trouble on rebasing and resolving conflicts, please feel free to mention me in this discussion and I am always ready to help!

**Comment by tyn1998** [comment]
Great job! Thanks!

/approve

**Comment by tyn1998** [comment]
close #276

**Pull request merged**